server.py

```diff
@@ -78,15 +80,29 @@ def set_connection_keys(self, conn):
             logger.debug("Not valid Noise communication - invalid length of public bytes.")
             return False
 
-    def receive_tpm_data(self, conn):
-        # TODO better storing client's tpm_data
-        self.tpm_data = conn.recv(constants.SOCK_BUFFER)
-        self.tpm_data = pickle.loads(self.tpm_data)
+    def validate_check_client_tpm(self, tpm_data: tuple) -> bool:
+        data, (ec_x, ec_y), (sig_r, sig_s) = tpm_data
+        if not ecdsa_validate(ec_x, ec_y, sig_r, sig_s, data):
+            logger.info("TPM data were validated successfully.")
+            return False
+        for user in self.user_list:
+            if user.check(data[-32:]):  # comparing PCR digest
+                return True
+        return False
+
+    def receive_tpm_data(self, conn: socket.socket) -> tuple:
+        tpm_data = conn.recv(constants.SOCK_BUFFER)
+        tpm_data = pickle.loads(tpm_data)
         logger.debug(f"TPM data has been received.")
+        return tpm_data
 
     def preshared_tpm_value(self, conn) -> bytes:
-        self.receive_tpm_data(conn)
-        data, _, _ = self.tpm_data
+        tpm_data = self.receive_tpm_data(conn)
+        if not self.validate_check_client_tpm(tpm_data):
+            conn.send(b"0")
```

Comment:
Maybe we could send send something descriptive, I believe now we are encoding ASCII '0' > 0x30
Maybe then OK/NOK?

Reply:
Could be. Or just send raw bytes. But why to send something descriptive when it is handled based on the T/F value?

Reply:
For demo purposes :D no reason, but it seemed easier than parsing raw bytes.